### DISCUSSION TITLE
Timestamps and isoDates

### DISCUSSION THREAD

**Discussion by acamaral** [sensedeep/dynamodb-onetable#310] [Q&A] [answered]
Hello, everyone.

I'm using `isoDates=true` in my schema definition (params), but I'd like to set it to _false_ for **timestamps** (_created_ and _updated_).

Can anybody help me with this?

Thanks in advance.

**Comment by mobsense** [comment, marked as answer]
Are you saying you want all Dates to be ISO strings except for timestamps where you want them to be Unix epochs?

Can you help me understand why this is an important use case for you?  I may be able to suggest alternatives.

**Reply by acamaral** [reply]
Thanks for your answer, @mobsense. I think dates stored as Unix epochs save space (less cost - let me know if I'm wrong), so I thought it would be nice to store all date attributes in Javascript ISO string, except the timestamps.

I'm trying SenseDeep, but I didn't see DynamoDB Studio yet. If it show date attributes in Javascript ISO string or any other "human readable" format, I will forget this and let DynamoDB stores all date attributes as Unix epochs.

**Reply by mobsense** [reply]
An ISO string is 24 bytes + string length, so it is less compact than a number date.

SenseDeep itself stores dates as epoch numbers, but if we had it again, we'd have selected ISO dates. The SenseDeep DynamoDB studio does not render dates in human readable form -- but that is a good feature request.

I think it is better to have all your dates in a consistent format. Excepting timestamps seems like it may cause more trouble than it is worth.

**Reply by shishkin** [reply]
I tend to prefer ISO dates for readability but recently discovered DynamoDB TTL feature that only works with Unix timestamps. Being able to specify schema field as TTL timestamp while using ISO for the rest of the Date fields by default would be useful.

**Reply by mobsense** [reply]
You could mark those TTL fields as Number. They are time in seconds, not milliseconds, so you cannot use a Date type regardless which is milliseconds.

**Reply by shishkin** [reply]
Maybe this could be a feature to allow users to use Date types on a column that would be turned into seconds TTL timestamp by OneTable.

**Reply by mobsense** [reply]
I'll create an enhancement issue for that suggestion.

**Reply by mobsense** [reply]
My bad. This already exists. 

See ttl in the doc: https://doc.onetable.io/api/table/schemas/attributes/

This will perform the following code when given a date. i.e. The date value in milliseconds will be converted to time in seconds
```
        if (field.ttl) {
            //  Convert dates to unix epoch in seconds
            if (value instanceof Date) {
                value = value.getTime()
            } else if (typeof value == 'string') {
                value = (new Date(Date.parse(value))).getTime()
            }
            value = Math.ceil(value / 1000)
```

**Reply by shishkin** [reply]
Good to know :D
